src/Utility/FieldFunctions.C

```diff
@@ -317,6 +317,54 @@ Real FieldFunction::evaluate (const Vec3& X) const
 }
 
 
+Vec3 FieldFunction::gradient (const Vec3& X) const
+{
+  if (pidx >= field.size() || !field[pidx])
+    return Vec3();
+
+  const Vec4* x4 = dynamic_cast<const Vec4*>(&X);
+  if (!x4)
+    return Vec3();
+
+  int level = this->findClosestLevel(x4->t);
+  if (level < 0) return Vec3();
+  if (level != currentLevel)
+    if (const_cast<FieldFunction*>(this)->load({fName},bName,level))
+      currentLevel = level;
+
+  Vector res;
+  field[pidx]->gradFE(ItgPoint(x4->u, x4->idx), res);
+  return res;
```

Comment:
In other methods in this file (e.g., `FieldsFuncBase::getValues()`) you have introduced `std::move()` to avoid unnecessary copying. So maybe here also, for the consistency? Otherwise it appears a bit random.

Reply:
can't move as we are using an implicit ctor here (Vector -> Vec3).